lib/ansible/modules/network/fortios/fortios_web_proxy_profile.py

```diff
@@ -0,0 +1,383 @@
+revision:
+  description: Internal revision number
+  returned: always
+  type: str
+  sample: "17.0.2.10658"
+serial:
+  description: Serial number of the unit
+  returned: always
+  type: str
+  sample: "FGVMEVYYQT3AB5352"
+status:
+  description: Indication of the operation's result
+  returned: always
+  type: str
+  sample: "success"
+vdom:
+  description: Virtual domain used
+  returned: always
+  type: str
+  sample: "root"
+version:
+  description: Version of the FortiGate
+  returned: always
+  type: str
+  sample: "v5.6.3"
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def login(data, fos):
+    host = data['host']
+    username = data['username']
+    password = data['password']
+
+    fos.debug('on')
+    if 'https' in data and not data['https']:
+        fos.https('off')
+    else:
+        fos.https('on')
+
+    fos.login(host, username, password)
+
+
+def filter_web_proxy_profile_data(json):
+    option_list = ['header-client-ip', 'header-front-end-https', 'header-via-request',
+                   'header-via-response', 'header-x-authenticated-groups', 'header-x-authenticated-user',
+                   'header-x-forwarded-for', 'headers', 'log-header-change',
+                   'name', 'strip-encoding']
+    dictionary = {}
+
+    for attribute in option_list:
+        if attribute in json and json[attribute] is not None:
+            dictionary[attribute] = json[attribute]
+
+    return dictionary
+
+
+def web_proxy_profile(data, fos):
+    vdom = data['vdom']
+    web_proxy_profile_data = data['web_proxy_profile']
+    filtered_data = filter_web_proxy_profile_data(web_proxy_profile_data)
+
+    if web_proxy_profile_data['state'] == "present":
+        return fos.set('web-proxy',
+                       'profile',
+                       data=filtered_data,
+                       vdom=vdom)
+
+    elif web_proxy_profile_data['state'] == "absent":
+        return fos.delete('web-proxy',
+                          'profile',
+                          mkey=filtered_data['name'],
+                          vdom=vdom)
+
+
+def fortios_web_proxy(data, fos):
+    login(data, fos)
+
+    if data['web_proxy_profile']:
+        resp = web_proxy_profile(data, fos)
+
+    fos.logout()
+    return not resp['status'] == "success", resp['status'] == "success", resp
+
+
+def main():
+    fields = {
+        "host": {"required": True, "type": "str"},
+        "username": {"required": True, "type": "str"},
+        "password": {"required": False, "type": "str", "no_log": True},
+        "vdom": {"required": False, "type": "str", "default": "root"},
+        "https": {"required": False, "type": "bool", "default": True},
+        "web_proxy_profile": {
+            "required": False, "type": "dict",
+            "options": {
+                "state": {"required": True, "type": "str",
+                          "choices": ["present", "absent"]},
+                "header-client-ip": {"required": False, "type": "str",
```

Comment:
For what it's worth, these names are technically invalid.  We are working on adding CI checks to ensure that arguments are valid python identifiers (variable names), which prohibits using hyphens in an argument name.

At the moment the tool we are using in CI doesn't recurse suboptions, only the top level, but will start validating the names of suboptions soon, at which point, these argument names will be classified as invalid.

Is there the potential that these could be converted now?

Reply:
Changing rules 2 weeks before a release is a sure method to loose contributors.

We agreed to fix those with some reasonable time to comply.  Can we log a bug and fix for 2.9 ?

In addition it is basically forcing us a rule on the naming of our own API.